lib/schemas.js

```diff
@@ -44,6 +44,26 @@ Schemas.prototype.getSchema = function (schemaId) {
   return this.store[schemaId]
 }
 
+function isCustomSchemaPrototype (schema) {
+  return typeof schema === 'object' && Object.getPrototypeOf(schema) !== Object.prototype
+}
+
+function hasCustomSchemaPrototype (routeSchemas) {
+  for (const key in routeSchemas) {
+    if (key === 'response') {
```

Comment:
We can't check the response like this.

This won't work if the user sets a json schema as `header` and a custom `response` schema: in this case the `header` schema will not be processed because the `response` check will be true and it will bypass all the following logic.

It is necessary to run this check with another implementation

Reply:
Hey @Eomm thanks for the feedback. This seems to be how it currently works: if any one schema is a custom class then the whole normalization is skipped. I can address that and add additional test cases if that's not the intended behavior.

Reply:
Yeah, for sure, it needs some refactoring.

We can sum up these cases:

input | output
---|---
json schema or empty checked | json schema checked
custom class checked | json schema checked
json schema or empty checked | custom class processed 🐛
custom class checked | custom class skipped 🐛

We need to fix only the two 🐛 cases without introducing other breaking 👍🏼